src/main/java/com/dnastack/ddap/explore/wes/controller/AccessController.java

```diff
@@ -0,0 +1,116 @@
+package com.dnastack.ddap.explore.wes.controller;
+
+import com.dnastack.ddap.common.security.UserTokenCookiePackager;
+import com.dnastack.ddap.explore.dam.client.ReactiveDamClient;
+import com.dnastack.ddap.explore.wes.client.AccessErrorException;
+import com.dnastack.ddap.explore.wes.service.ViewsService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.server.reactive.ServerHttpRequest;
+import org.springframework.http.server.reactive.ServerHttpResponse;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.util.UriComponentsBuilder;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import dam.v1.DamService.GetTokenResponse;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.*;
+
+import static com.dnastack.ddap.common.security.UserTokenCookiePackager.CookieKind;
+
+
+@Slf4j
+@RestController
+@RequestMapping("/api/v1alpha/{realm}/access")
+public class AccessController {
+
+    private final UserTokenCookiePackager cookiePackager;
+    private Map<String, ReactiveDamClient> damClients;
+    private final ViewsService viewsService;
+
+    @Autowired
+    public AccessController(UserTokenCookiePackager cookiePackager,
+                            Map<String, ReactiveDamClient> damClients,
+                            ViewsService viewsService) {
+        this.cookiePackager = cookiePackager;
+        this.damClients = damClients;
+        this.viewsService = viewsService;
+    }
+
+    @GetMapping("/{bucketName}/{*filePath}")
```

Comment:
`{*filePath}` iis this a valid syntax? can you explain/point me to docs where the asterix character is explained?

Reply:
It is a valid syntax, doc for that https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/web/util/pattern/PathPattern.html

Reply:
It is a valid syntax, doc for that https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/web/util/pattern/PathPattern.html